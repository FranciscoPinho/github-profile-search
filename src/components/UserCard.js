import React from "react";
import { Icon, Card, Image } from "semantic-ui-react";

const ItemLine = ({ icon, item, link = false }) =>
  link ? (
    <a href={item} target="_blank" rel="noopener noreferrer">
      <Icon name={icon} /> {item}{" "}
    </a>
  ) : (
    <p>
      <Icon name={icon} /> {item}{" "}
    </p>
  );

const UserCard = ({ user }) =>
  user ? (
    <Card>
      {user.profileImage && (
        <Image src={user.profileImage} wrapped ui={false} />
      )}
      <Card.Content>
        {user.name && <Card.Header>{user.name}</Card.Header>}
        <Card.Description>
          {user.location && (
            <ItemLine icon="map marker alternate" item={user.location} />
          )}
          {user.email && <ItemLine icon="mail" item={user.email} />}
          {user.profileLink && (
            <ItemLine icon="linkify" link item={user.profileLink} />
          )}
        </Card.Description>
      </Card.Content>
    </Card>
  ) : null;

export default UserCard;
