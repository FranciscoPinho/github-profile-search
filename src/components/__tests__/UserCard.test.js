import React from "react";
import { shallow, mount } from "enzyme";
import UserCard from "../UserCard";

const user = {
  name: "FranciscoPinho",
  profileLink: "testLink",
  profileImage: "https://identicons.github.com/jasonlong.png",
  email: "example@gmail.com",
  location: "Porto, Portugal"
};

describe("UserCard should render correctly", () => {
  it("with user", () => {
    const component = mount(<UserCard user={user} />);

    expect(component).toMatchSnapshot();
  });
  it("as null with no user", () => {
    const component = shallow(<UserCard />);

    expect(component).toMatchSnapshot();
  });
  it("with user without name attribute", () => {
    const component = shallow(
      <UserCard user={{ ...user, name: null }} />
    );

    expect(component).toMatchSnapshot();
  });
  it("with user without profileLink attribute", () => {
    const component = shallow(
      <UserCard user={{ ...user, profileLink: null }} />
    );

    expect(component).toMatchSnapshot();
  });
  it("with user without profileImage attribute", () => {
    const component = shallow(
      <UserCard user={{ ...user, profileImage: null }} />
    );

    expect(component).toMatchSnapshot();
  });
  it("with user without email attribute", () => {
    const component = shallow(
      <UserCard user={{ ...user, email: null }} />
    );

    expect(component).toMatchSnapshot();
  });
  it("with user without location attribute", () => {
    const component = shallow(
      <UserCard user={{ ...user, location: null }} />
    );

    expect(component).toMatchSnapshot();
  });
});
