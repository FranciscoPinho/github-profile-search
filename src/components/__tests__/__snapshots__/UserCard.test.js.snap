// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should render correctly as null with no user 1`] = `""`;

exports[`UserCard should render correctly with user 1`] = `
<UserCard
  user={
    Object {
      "email": "example@gmail.com",
      "location": "Porto, Portugal",
      "name": "FranciscoPinho",
      "profileImage": "https://identicons.github.com/jasonlong.png",
      "profileLink": "testLink",
    }
  }
>
  <Card>
    <div
      className="ui card"
      onClick={[Function]}
    >
      <Image
        as="img"
        src="https://identicons.github.com/jasonlong.png"
        ui={false}
        wrapped={true}
      >
        <div
          className="image"
        >
          <img
            src="https://identicons.github.com/jasonlong.png"
          />
        </div>
      </Image>
      <CardContent>
        <div
          className="content"
        >
          <CardHeader>
            <div
              className="header"
            >
              FranciscoPinho
            </div>
          </CardHeader>
          <CardDescription>
            <div
              className="description"
            >
              <ItemLine
                icon="map marker alternate"
                item="Porto, Portugal"
              >
                <p>
                  <Icon
                    as="i"
                    name="map marker alternate"
                  >
                    <i
                      aria-hidden="true"
                      className="map marker alternate icon"
                      onClick={[Function]}
                    />
                  </Icon>
                   
                  Porto, Portugal
                   
                </p>
              </ItemLine>
              <ItemLine
                icon="mail"
                item="example@gmail.com"
              >
                <p>
                  <Icon
                    as="i"
                    name="mail"
                  >
                    <i
                      aria-hidden="true"
                      className="mail icon"
                      onClick={[Function]}
                    />
                  </Icon>
                   
                  example@gmail.com
                   
                </p>
              </ItemLine>
              <ItemLine
                icon="linkify"
                item="testLink"
                link={true}
              >
                <a
                  href="testLink"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <Icon
                    as="i"
                    name="linkify"
                  >
                    <i
                      aria-hidden="true"
                      className="linkify icon"
                      onClick={[Function]}
                    />
                  </Icon>
                   
                  testLink
                   
                </a>
              </ItemLine>
            </div>
          </CardDescription>
        </div>
      </CardContent>
    </div>
  </Card>
</UserCard>
`;

exports[`UserCard should render correctly with user without email attribute 1`] = `
<Card>
  <Image
    as="img"
    src="https://identicons.github.com/jasonlong.png"
    ui={false}
    wrapped={true}
  />
  <CardContent>
    <CardHeader>
      FranciscoPinho
    </CardHeader>
    <CardDescription>
      <ItemLine
        icon="map marker alternate"
        item="Porto, Portugal"
      />
      <ItemLine
        icon="linkify"
        item="testLink"
        link={true}
      />
    </CardDescription>
  </CardContent>
</Card>
`;

exports[`UserCard should render correctly with user without location attribute 1`] = `
<Card>
  <Image
    as="img"
    src="https://identicons.github.com/jasonlong.png"
    ui={false}
    wrapped={true}
  />
  <CardContent>
    <CardHeader>
      FranciscoPinho
    </CardHeader>
    <CardDescription>
      <ItemLine
        icon="mail"
        item="example@gmail.com"
      />
      <ItemLine
        icon="linkify"
        item="testLink"
        link={true}
      />
    </CardDescription>
  </CardContent>
</Card>
`;

exports[`UserCard should render correctly with user without name attribute 1`] = `
<Card>
  <Image
    as="img"
    src="https://identicons.github.com/jasonlong.png"
    ui={false}
    wrapped={true}
  />
  <CardContent>
    <CardDescription>
      <ItemLine
        icon="map marker alternate"
        item="Porto, Portugal"
      />
      <ItemLine
        icon="mail"
        item="example@gmail.com"
      />
      <ItemLine
        icon="linkify"
        item="testLink"
        link={true}
      />
    </CardDescription>
  </CardContent>
</Card>
`;

exports[`UserCard should render correctly with user without profileImage attribute 1`] = `
<Card>
  <CardContent>
    <CardHeader>
      FranciscoPinho
    </CardHeader>
    <CardDescription>
      <ItemLine
        icon="map marker alternate"
        item="Porto, Portugal"
      />
      <ItemLine
        icon="mail"
        item="example@gmail.com"
      />
      <ItemLine
        icon="linkify"
        item="testLink"
        link={true}
      />
    </CardDescription>
  </CardContent>
</Card>
`;

exports[`UserCard should render correctly with user without profileLink attribute 1`] = `
<Card>
  <Image
    as="img"
    src="https://identicons.github.com/jasonlong.png"
    ui={false}
    wrapped={true}
  />
  <CardContent>
    <CardHeader>
      FranciscoPinho
    </CardHeader>
    <CardDescription>
      <ItemLine
        icon="map marker alternate"
        item="Porto, Portugal"
      />
      <ItemLine
        icon="mail"
        item="example@gmail.com"
      />
    </CardDescription>
  </CardContent>
</Card>
`;
